actions/copy/worker.js

```diff
@@ -41,7 +41,7 @@ async function main(params) {
             logger.error(payload);
         } else if (!spToken || !adminPageUri) {
             payload = 'Required data is not available to proceed with FG Copy action.';
-            updateStatusToStateLib(projectPath, PROJECT_STATUS.COMPLETED_WITH_ERROR, payload, undefined, COPY_ACTION);
+            updateStatusToStateLib(projectPath, PROJECT_STATUS.FAILED, payload, undefined, COPY_ACTION);
```

Comment:
We might have to set it to `FAILED` in the catch block (line 67) as well as we do not know whether it completed at that step, correct?

Same question for the catch in promote worker.js

Reply:
@sukamat  - Status handling for capturing error and status thrown  at ln157 but that is one use case. There are a couple more being thrown from sharepoint.js for updating excel(ln-278) and creating folders(ln-105) that might end up being FAILED or COMPLETED WITH ERROR. My intention was not to mess up with the core functions to pass all these params to capture status at those locations and wanted to capture at one place to keep it clean.

Reply:
OK. My only concern is that it can represent the wrong status. 

COMPLETED_WITH_ERROR makes sense if there were issues when updating the excel file or if copy or preview calls on certain pages failed but it went through the rest of the process fine. But if it errors out at other places and the process stops at the catch block, then we don't want to set the status as COMPLETED_WITH_ERROR.

May be it can be a separate ticket to handle this case if it going to need changes at multiple places?

Reply:
@sukamat a new ticket makes sense but if we can display error message from our status fetch on a tooltip that would give authors/dev a better idea of where it failed.

Reply:
Good idea. I will log a ticket to handle that separately.